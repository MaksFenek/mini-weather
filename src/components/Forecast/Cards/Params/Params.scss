.params {
  ul {
    list-style: none;
    li {
      display: flex;
      align-items: center;
      font-size: 20px;
      margin-bottom: 24px;
      span {
        margin-left: 12px;
      }
    }
  }
}
