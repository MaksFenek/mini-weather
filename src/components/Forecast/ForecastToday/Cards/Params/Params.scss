.params {
  ul {
    list-style: none;
    li {
      display: flex;
      align-items: center;
      font-size: 20px;
      margin-bottom: 24px;
      p {
        margin-left: 12px;
        span {
          font-weight: bold;
        }
      }
    }
  }
}
