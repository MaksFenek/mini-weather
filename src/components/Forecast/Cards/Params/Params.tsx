import React from 'react';
import './Params.scss';

interface IParams {
  humidity: number | '--';
  pressure: number | '--';
  wind: {
    speed: number;
    deg: number;
  };
}

const Params = ({ humidity = '--', pressure = '--', wind }: IParams) => {
  return (
    <div className='forecast-content__params'>
      <div className='params'>
        <ul>
          <li>
            <svg className='icon' height='36px' width='36px'>
              <use xlinkHref={`/src/Icons/params.svg#humidity`} />
            </svg>
            <span>Humidity: {humidity} %</span>
          </li>
          <li>
            <svg className='icon' height='36px' width='36px'>
              <use xlinkHref={`/src/Icons/params.svg#barometer`} />
            </svg>
            <span>Pressure: {pressure} hPa</span>
          </li>
          <li>
            <svg
              className='icon'
              height='52px'
              width='52px'
              style={{ margin: '-8px' }}
            >
              <use
                xlinkHref={`/src/Icons/compass.svg#main
              `}
              ></use>
              <use
                style={{
                  transform: `rotate(${wind.deg + 90}deg)`,
                  transformOrigin: 'center',
                }}
                xlinkHref={`/src/Icons/compass.svg#arrow`}
              ></use>
            </svg>
            <span>Wind: {wind ? wind.speed : '--'} mph</span>
          </li>
        </ul>
      </div>
    </div>
  );
};

export default Params;
