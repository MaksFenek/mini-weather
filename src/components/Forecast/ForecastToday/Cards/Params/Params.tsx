// React
import React from 'react';

// Styles
import './Params.scss';

// Types
interface IParams {
  humidity: number | '--';
  pressure: number | '--';
  wind: {
    speed: number;
    deg: number;
  };
}

const Params: React.FC<IParams> = ({
  humidity = '--',
  pressure = '--',
  wind,
}) => {
  return (
    <div className='params'>
      <ul>
        <li>
          <svg className='icon' height='36px' width='36px'>
            <use xlinkHref={`icons/params.svg#humidity`} />
          </svg>
          <p>
            Humidity: <span>{humidity}</span> %
          </p>
        </li>
        <li>
          <svg className='icon' height='36px' width='36px'>
            <use xlinkHref={`icons/params.svg#barometer`} />
          </svg>
          <p>
            Pressure: <span>{pressure}</span> hPa
          </p>
        </li>
        <li>
          <svg
            className='icon'
            height='52px'
            width='52px'
            style={{ margin: '-7px' }}
          >
            <use
              xlinkHref={`icons/compass.svg#main
              `}
            ></use>
            <use
              style={{
                transform: `rotate(${wind.deg + 270}deg)`,
                transformOrigin: 'center',
              }}
              xlinkHref={`icons/compass.svg#arrow`}
            ></use>
          </svg>
          <p>
            Wind: <span>{wind ? wind.speed : '--'}</span> mph
          </p>
        </li>
      </ul>
    </div>
  );
};

export default Params;
