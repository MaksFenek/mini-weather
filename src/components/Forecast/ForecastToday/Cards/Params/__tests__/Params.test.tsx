import React from 'react';
import Params, { IParams } from '../Params';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

describe('<Params/>', () => {
  const props: IParams = {
    humidity: 40,
    pressure: 1000,
    wind: {
      speed: 2.3,
      deg: 120,
    },
  };

  const wrapper = mount(<Params {...props} />);
  const tree = renderer.create(<Params {...props} />).toJSON();

  it('should render correctly', () => {
    expect(wrapper.find(`[aria-label='humidity']`).text()).toEqual(
      `${props.humidity}`
    );
    expect(wrapper.find(`[aria-label='pressure']`).text()).toEqual(
      `${props.pressure}`
    );
    expect(wrapper.find(`[aria-label='wind']`).text()).toEqual(
      `${props.wind.speed}`
    );
    expect(tree).toMatchSnapshot();
  });
});
