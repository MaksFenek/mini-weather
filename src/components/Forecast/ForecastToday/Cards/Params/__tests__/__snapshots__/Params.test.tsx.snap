// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Params/> should render correctly 1`] = `
<div
  className="params"
>
  <ul>
    <li>
      <svg
        className="icon"
        height="36px"
        width="36px"
      >
        <use
          xlinkHref="icons/params.svg#humidity"
        />
      </svg>
      <p>
        Humidity: 
        <span
          aria-label="humidity"
        >
          40
        </span>
         %
      </p>
    </li>
    <li>
      <svg
        className="icon"
        height="36px"
        width="36px"
      >
        <use
          xlinkHref="icons/params.svg#barometer"
        />
      </svg>
      <p>
        Pressure: 
        <span
          aria-label="pressure"
        >
          1000
        </span>
         hPa
      </p>
    </li>
    <li>
      <svg
        className="icon"
        height="52px"
        style={
          Object {
            "margin": "-7px",
          }
        }
        width="52px"
      >
        <use
          xlinkHref="icons/compass.svg#main
              "
        />
        <use
          style={
            Object {
              "transform": "rotate(390deg)",
              "transformOrigin": "center",
            }
          }
          xlinkHref="icons/compass.svg#arrow"
        />
      </svg>
      <p>
        Wind: 
        <span
          aria-label="wind"
        >
          2.3
        </span>
         mph
      </p>
    </li>
  </ul>
</div>
`;
